src/CodeGen_GPU_Host.cpp

```diff
@@ -1,13 +1,23 @@
 #include <sstream>
 
-#include "CodeGen_D3D12Compute_Dev.h"
+#ifdef WITH_D3D12
```

Comment:
I think it would be better to move these into the header guards so that they may be included unconditionally.